File: app-android/src/main/java/org/mtransit/android/analytics/IAnalyticsManager.java
package org.mtransit.android.analytics;

import androidx.annotation.MainThread;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.Size;

import org.mtransit.android.ui.view.common.IActivity;

@SuppressWarnings("unused")
public interface IAnalyticsManager {

	void setUserProperty(@NonNull @Size(min = 1L, max = 24L) String name, int value);

	void setUserProperty(@NonNull @Size(min = 1L, max = 24L) String name, @NonNull String value);

	void logEvent(@NonNull @Size(min = 1L, max = 40L) String name);

	void logEvent(@NonNull @Size(min = 1L, max = 40L) String name, @Nullable AnalyticsEventsParamsProvider params);

	@MainThread
	void trackScreenView(@NonNull IActivity activity, @NonNull Trackable page);

	interface Trackable {
		@NonNull
		String getScreenName();
	}
}

File: app-android/src/main/java/org/mtransit/android/analytics/IAnalyticsManager.kt
package org.mtransit.android.analytics

import androidx.annotation.MainThread
import androidx.annotation.Size

@Suppress("unused")
interface IAnalyticsManager {
    fun setUserProperty(@Size(min = 1L, max = 24L) name: String, value: Int)

    fun setUserProperty(@Size(min = 1L, max = 24L) name: String, value: String)

    fun logEvent(@Size(min = 1L, max = 40L) name: String) = logEvent(name, params = null)

    fun logEvent(@Size(min = 1L, max = 40L) name: String, params: AnalyticsEventsParamsProvider?)

    @MainThread
    fun trackScreenView(page: AnalyticsScreen)

    @MainThread
    fun trackButtonClick(buttonName: String, page: AnalyticsScreen?) =
        trackButtonClick(buttonName, null, page)

    @MainThread
    fun trackButtonClick(buttonName: String, buttonId: String? = null, page: AnalyticsScreen? = null)
}
